### PR TITLE
converted slashcommand-leave coffee to js

### PR DESCRIPTION
**Pull request by ggazzo**
@RocketChat/core

### CODE REVIEW

File: packages/rocketchat-slashcommands-leave/leave.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
+/*
+ * Leave is a named function that will replace /leave commands
+ * @param {Object} message - The message object
+ */
+	if (Meteor.isClient) {
+		RocketChat.slashCommands.add('leave', null, {
```
Comment (rodrigok) [inline packages/rocketchat-slashcommands-leave/leave.js]: Why did you change `undefined` to `null`?

File: packages/rocketchat-slashcommands-leave/leave.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+			description: 'Leave_the_current_channel'
+		});
+		RocketChat.slashCommands.add('part', null, {
+			description: 'Leave_the_current_channel'
+		});
+	} else {
+		const Leave = function Leave(command, params, item) {
```
Comment (rodrigok) [inline packages/rocketchat-slashcommands-leave/leave.js]: Why declare a variable if you are naming a function?

File: packages/rocketchat-slashcommands-leave/leave.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+			if (command !== 'leave' || command !== 'part') {
```
Comment (rodrigok) [inline packages/rocketchat-slashcommands-leave/leave.js]: If you invert a condition you need to invert the conditionals too:
```
if(command == "leave" || command == "part")
  continue
```
Should be:
```
if (command !== 'leave' && command !== 'part') {
  return
```

In your conde command `part` will execute the `return` cuz it's different from `leave`

File: packages/rocketchat-slashcommands-leave/leave.js (hunk 1 above)
Comment (ggazzo) [inline packages/rocketchat-slashcommands-leave/leave.js]: because in this case, there are no difference.
the test is ```RocketChat.slashCommands.commands[command]?.callback?``` it will test ```callback != null;```
, but I can change if you prefer.

Comment (graywolf336) [inline packages/rocketchat-slashcommands-leave/leave.js]: To me `undefined` reads and is more "javascripty" than `null`, so I would personally prefer `undefined`.

File: packages/rocketchat-slashcommands-leave/leave.js (hunk 3 above)
Comment (ggazzo) [inline packages/rocketchat-slashcommands-leave/leave.js]: my mistake, sorry :/
